### ISSUE TITLE
questions about code (class UnmaskingLlamaModel)

### ISSUE THREAD

**Issue by maxh2018** [4AI/LS-LLaMA#24]
![1728398565381](https://github.com/user-attachments/assets/b4c4d9d0-a05a-4dd3-85ef-b43ea3b29056)
---------------------------------------------------------------------------------------------------------------------
code 1:
 if attention_mask is None:
            attention_mask = torch.ones(
                (batch_size, seq_length_with_past), dtype=torch.bool, device=inputs_embeds.device
            )
---------------------------------------------------------------------------------------------------------------------
code2:
        # causal mask
        '''
        attention_mask = self._prepare_decoder_attention_mask(
            attention_mask, (batch_size, seq_length), inputs_embeds, past_key_values_length
        )
        print('unmasking attention mask:')
        print(attention_mask)
        '''
---------------------------------------------------------------------------------------------------------------------
code3:
        # remove causal mask
        attention_mask = torch.zeros(
            (batch_size, 1, seq_length, seq_length), device=inputs_embeds.device
        )

whats the meaning of code3? And why 'torch.zeros' rather than 'torch.ones' as in code1?
